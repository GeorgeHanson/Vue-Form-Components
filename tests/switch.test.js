import Vue from 'vue';
import SwitchComponent from '../src/Components/Switch.vue';

describe('SwitchComponent', () => {
    test("it generates the markup", () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            render: (createElement) => {
                return createElement(
                    SwitchComponent,
                    {
                        props: {
                            name: 'marketing',
                            label: 'Would you like to receive marketing updates?'
                        }
                    }
                )
            },
        });

        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});