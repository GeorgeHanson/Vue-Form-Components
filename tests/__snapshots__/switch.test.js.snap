// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchComponent it generates the markup 1`] = `
< div data - server - rendered = "true"
class = "form-group switch-group" > < div class = "col-sm-4" > < label class = "control-label" > Would you like to receive marketing updates ? < /label> <!----></div > < div class = "col-sm-8" > < label class = "switch-component" > < input type = "checkbox"
name = "marketing" > < div class = "slider round" > <!----> <!----></div></label></div></div>
`;
